source/index.ts

```diff
@@ -1,28 +1,16 @@
 import callsites from 'callsites';
 import {inferLabel} from './lib/utils/infer-label';
 import {Predicate} from './lib/predicates/predicate';
-import {AnyPredicate} from './lib/predicates/any';
 import {testSymbol, BasePredicate, isPredicate} from './lib/predicates/base-predicate';
-import {StringPredicate} from './lib/predicates/string';
-import {NumberPredicate} from './lib/predicates/number';
-import {BooleanPredicate} from './lib/predicates/boolean';
-import {ArrayPredicate} from './lib/predicates/array';
-import {ObjectPredicate} from './lib/predicates/object';
-import {DatePredicate} from './lib/predicates/date';
-import {ErrorPredicate} from './lib/predicates/error';
-import {MapPredicate} from './lib/predicates/map';
-import {WeakMapPredicate} from './lib/predicates/weak-map';
-import {SetPredicate} from './lib/predicates/set';
-import {WeakSetPredicate} from './lib/predicates/weak-set';
-
-type TypedArray = Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array;
+import modifiers, {Modifiers} from './modifiers';
+import predicates, {Predicates} from './predicates';
 
 /**
  * @hidden
  */
 export type Main = <T>(value: T, label: string | Function, predicate: BasePredicate<T>) => void;
 
-export interface Ow {
+export interface Ow extends Modifiers, Predicates {
```

Comment:
Technically this isn't necessary because the `modifiers` and `predicates` loaders at line 98 will return the correct type. The reason I added this is so that the documentation being generated looks the same as what we have right now.

Reply:
Can you mention this in a code comment?

Reply:
Added it, something like that?

Reply:
Worth opening an issue on TypeDoc about better inference for the docs?

Reply:
I'm not sure. The thing is that TSDoc does it kind of correct. It just documents to 3 interfaces separately. It documents `Ow`, `Predicates` and `Modifiers`. Because `Ow` is now only holding the base function, `isValid` and `create`, the other properties aren't present in the interface and thus doesn't end up under the `Ow` interface in the documentation. Which feels weird to me and thus I ended up doing what I did, extending from both interfaces explicitely.

So not really sure if this is the fault of TSDoc.